crates/teloxide/src/dispatching/dispatcher.rs

```diff
@@ -495,6 +495,32 @@ impl<R, Err, Key> Dispatcher<R, Err, Key> {
     }
 }
 
+/// Extension methods for adding `tracing` to [`UpdateHandler`].
+///
+/// This trait is only available if the `tracing` feature is enabled.
+#[cfg(feature = "tracing")]
+pub trait UpdateHandlerTracingExt<E> {
+    /// Returns an `UpdateHandler` with tracing enabled.
+    fn with_tracing(self, f: fn(Arc<Update>) -> tracing::Span) -> Self;
+}
+
+#[cfg(feature = "tracing")]
+impl<E: 'static> UpdateHandlerTracingExt<E> for UpdateHandler<E> {
+    fn with_tracing(self, f: fn(Arc<Update>) -> tracing::Span) -> Self {
+        use dptree::HandlerDescription;
+        use tracing::Instrument;
+
+        // This is a hacky replacement for `handler.description().clone()`.
+        // Ideally cloning `DpHandlerDescription` would be supported by `teloxide`.
+        let description = DpHandlerDescription::entry().merge_chain(self.description());
```

Comment:
Please leave a `FIXME` comment here, so we can fix this later.

Reply:
Added it.